tests/components/home_plus_control/test_switch.py

```diff
@@ -0,0 +1,456 @@
+"""Test the Legrand Home+ Control switch platform."""
+import datetime as dt
+from unittest.mock import patch
+
+from homepluscontrol.homeplusapi import HomePlusControlApiError
+
+from homeassistant import config_entries, setup
+from homeassistant.components.home_plus_control.const import (
+    CONF_SUBSCRIPTION_KEY,
+    DOMAIN,
+    ENTITY_UIDS,
+)
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+
+from tests.common import async_fire_time_changed
+from tests.components.home_plus_control.conftest import (
+    CLIENT_ID,
+    CLIENT_SECRET,
+    SUBSCRIPTION_KEY,
+)
+
+
+def entity_assertions(
+    hass,
+    num_exp_entities,
+    num_exp_devices=None,
+    expected_entities=None,
+    expected_devices=None,
+):
+    """Assert number of entities and devices."""
+    entity_reg = hass.helpers.entity_registry.async_get(hass)
+    device_reg = hass.helpers.device_registry.async_get(hass)
+
+    if num_exp_devices is None:
+        num_exp_devices = num_exp_entities
+
+    assert len(entity_reg.entities.keys()) == num_exp_entities
+    assert len(device_reg.devices.keys()) == num_exp_devices
+
+    if expected_entities is not None:
+        for exp_entity, present in expected_entities.items():
+            assert bool(entity_reg.async_get(exp_entity)) == present
+
+    if expected_devices is not None:
+        for exp_device, present in expected_devices.items():
+            assert bool(device_reg.async_get(exp_device)) == present
+
+
+def one_entity_assertion(hass, device_uid, availability):
+    """Assert the presence of an entity and its specified availability."""
+    entity_reg = hass.helpers.entity_registry.async_get(hass)
+    device_reg = hass.helpers.device_registry.async_get(hass)
+
+    device_id = device_reg.async_get_device({(DOMAIN, device_uid)}).id
+    device_entities = hass.helpers.entity_registry.async_entries_for_device(
+        entity_reg, device_id
+    )
+
+    assert len(device_entities) == 1
+    one_entity = device_entities[0]
+    assert (
+        hass.data["entity_platform"][DOMAIN][0].entities[one_entity.entity_id].available
```

Comment:
Don't assert hass.data. If the entity is unavailable it will be represented as `"unavailable"` state in the state machine.